licensing/src/main/java/io/crate/license/LicenseService.java

```diff
@@ -168,7 +196,22 @@ public void clusterChanged(ClusterChangedEvent event) {
 
         if (newLicenseKey != null && !newLicenseKey.equals(previousLicenseKey)) {
             try {
-                this.currentLicense.set(licenseData(decodeLicense(newLicenseKey)));
+                DecryptedLicenseData decryptedLicenseData = licenseData(decodeLicense(newLicenseKey));
+                LicenseExpiryNotification expiryNotification = getLicenseExpiryNotification(decryptedLicenseData);
+
+                if (expiryNotification != null) {
+                    long millisToExpiration = decryptedLicenseData.millisToExpiration();
+                    if (expiryNotification.equals(LicenseExpiryNotification.EXPIRED)) {
+                        logger.error(expiryNotification.notificationMessage(millisToExpiration));
+                        throw new InvalidLicenseException("License expired");
```

Comment:
Does it make sense to raise exceptions within `clusterChanged` ? The `ClusterApplierService` will just log any exception again - so we'd have duplicate log entries due to this

Reply:
I thought of this to be temporary as we need to do more stuff when the license is expired  (eg. disable operations). I'll just log it here and then we'll see how we handle monitoring the license (the expiration could be a scheduled event maybe?)

Reply:
done